### PR TITLE
Bugfix of issues related to resultfield definition and processing

### PR DESCRIPTION
**Pull request by LukasFehring**
## Description
I finished 3 issues related to resultfield definition and processing that caused problems for users.

## Motivation and Context

- `PyExperimenter`with invalid configuration files have terminated with an error, without giving further information about the issue (#139).
- Processing results of `PyExperimenter` terminated with an error if a given resultfield was not defined in the configuration file and therefore was not covered in the database. Unfortunately, no information was given which field caused the problem (#140).
- Resultfields have been mandatory, preventing users to execute experiments without the need of resultfields (#89)

#### Type Of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How has This Been Tested?

- Database provider: MySQL and SQLite
- Python version: 3.9.12
- Operating System: Ubuntu, MacOSX

Executed testcases, added minor testcases and executed examples. Also double checked that the checks for the existence of result/logtable columns work, by adapting `own_functions` in the notebook

All tests have been executed.

## Does this Close/Impact Existing Issues?

- Issue #139 
- Issue #140 
- Issue #89 

## Checklist

- [x] My change is based on the latest stage of the `develop` branch.
- [x] My change required a change of the documentation, which has been done.
- [x] I checked that the documentation can be build, visualizes everything as expected, and does not contain any warnings.
- [x] I have added/adapted tests to cover my changes.
- [x] The tests can be executed successfully.
- [x] The notebooks can be executed successfully.
- [x] The notebooks can be executed with `mysql` as provider.
- [x] I have added a description of the changes to `CHANGELOG.rst`.

### CODE REVIEW

File: py_experimenter/experimenter.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def __init__(self,
 
         self.database_credential_file_path = database_credential_file_path
         if not self._is_valid_configuration(self.config, database_credential_file_path):
-            raise InvalidConfigError('Invalid configuration')
+            raise InvalidConfigError('Invalid configuration. See log messages for details')
```
Comment (tornede) [inline py_experimenter/experimenter.py:102]: 'Invalid configuration, see previous log messages for details.'

File: py_experimenter/result_processor.py [hunk 2]
```diff
@@ -31,21 +30,26 @@ def __init__(self, config: ConfigParser, use_codecarbon: bool, codecarbon_config
...
+    def process_results(self, results: Dict) -> None:
         """
         Process results from the experiment and write them to the database. You can call this method, whenever you
         want to write results to the database.
         :param results: Dictionary with result field name and result value pairs.
         """
         if not self._valid_result_fields(list(results.keys())):
             invalid_result_keys = set(list(results.keys())) - set(self._result_fields)
-            raise InvalidResultFieldError(f"Invalid result keys: {invalid_result_keys}")
+            logging.error(f"The resultsfileds `{','.join(invalid_result_keys)}` are invalid.")
```
Comment (tornede) [inline py_experimenter/result_processor.py:51]: Invalid wrt what? They probably don't exists in the DB and therefore do not match with the experiment configuration at the time the DB was set up. Would be good to have this information given here.

**Review by tornede — CHANGES_REQUESTED** [review]
LGTM except the log message changes mentioned above

File: py_experimenter/result_processor.py (hunk 2 above)
Comment (LukasFehring) [reply py_experimenter/result_processor.py:51]: I changed this

File: py_experimenter/experimenter.py (hunk 1 above)
Comment (LukasFehring) [reply py_experimenter/experimenter.py:102]: I changed this

**Review by tornede — CHANGES_REQUESTED** [review]

**Review by tornede — APPROVED** [review]
LGTM

**Pull request merged**